app/controllers/api/groups_controller.rb

```diff
@@ -281,6 +281,30 @@ def update_marking_state
       end
     end
 
+    def add_tag
+      tag = Tag.find_by(id: params[:tag_id])
+      grouping = Grouping.find_by(id: params[:grouping_id])
```

Comment:
The other methods here use `self.grouping` (there's a helper method defined at the bottom of this file), and you should do the same here. Note that this requires a different configuration of params (`assignment_id` to specify the assignment, and then a `Group` id (not `Grouping` id), which is used to define the `record` for the controller

Reply:
I'm having an issue with this where I am sending ```id: group.id``` to the controller and the record is still nil resulting in self.grouping throwing an error. I have also tried sending ```group_id: group.id```. Do you know what might be going wrong?

I'm not quite sure what is going wrong. Will push for visibility.

Reply:
@WeissSa there's a problem with the tests. Look carefully at the definitions of the group and grouping factories (their definitions tell you what attributes you must/can specify when creating them in the test files).

Reply:
@david-yz-liu sorry that was a typo in my ```add_tag``` tests. I will definitely fix it, but the issue is that I get the message:
```Api::GroupsController An authenticated request requesting PUT add_tag should throw a 404 if the tag id is not found
     Failure/Error: @grouping ||= record.grouping_for_assignment(assignment.id)

     NoMethodError:
       undefined method `grouping_for_assignment' for nil:NilClass```
When running my remove_tag (or my newly updated add_tag)
I have tried using both id, and group_id to set the record so I am unsure why it is nil. Thanks!

Reply:
I would set a breakpoint somewhere to check the params manually. If `record` is `nil` then likely the id in the params is incorrect, possibly because the test objects are not being successfully created. And again, review the factory files.